src/server/game/Entities/Object/Object.cpp

```diff
@@ -568,42 +564,35 @@ void Object::BuildMovementUpdate(ByteBuffer* data, uint32 flags) const
         if (hasUnk3)
             data->WriteBit(0);
 
+        data->FlushBits();
+
         if (hasAreaTriggerSpline)
         {
             std::vector<G3D::Vector3> const& splinePoints = areaTrigger->GetSpline().getPoints();
 
-            *data << float(areaTriggerMiscTemplate->TimeToTarget);
-            *data << float(areaTrigger->GetTimeSinceCreated()); // ElapsedTimeForMovement
+            *data << uint32(areaTrigger->GetTimeToTarget());
+            *data << uint32(areaTrigger->GetElapsedTimeForMovement());
 
-            data->FlushBits();
             data->WriteBits(splinePoints.size(), 16);
 
             for (G3D::Vector3 const& spline : splinePoints)
-            {
-                *data << spline.x;
-                *data << spline.y;
-                *data << spline.z;
-            }
+                *data << spline;
         }
 
         if (hasTargetRollPitchYaw)
-        {
-            *data << float(0); // X
-            *data << float(0); // Y
-            *data << float(0); // Z
-        }
+            *data << G3D::Vector3::zero();
 
         if (hasScaleCurveID)
-            *data << int32(areaTriggerMiscTemplate->ScaleCurveId);
+            *data << uint32(areaTriggerMiscTemplate->ScaleCurveId);
```

Comment:
@joschiwald curve ids are signed in packet (not that it really matters, but i would like to preserve original structure signs)